src/util/metrics.rs

```diff
@@ -21,3 +24,49 @@ lazy_static! {
             &["type"]
         ).unwrap();
 }
+
+/// `run_prometheus` runs a background prometheus client.
+pub fn run_prometheus(
+    interval: Duration,
+    address: &str,
+    job: &str,
+) -> Option<thread::JoinHandle<()>> {
+    if interval == Duration::from_secs(0) {
+        return None;
+    }
+
+    let job = job.to_owned();
+    let address = address.to_owned();
+    let handler = thread::Builder::new()
+        .name("promepusher".to_owned())
+        .spawn(move || loop {
+            let metric_familys = prometheus::gather();
+
+            let res = prometheus::push_metrics(
+                &job,
+                prometheus::hostname_grouping_key(),
+                &address,
+                metric_familys,
+            );
+            if let Err(e) = res {
+                error!("fail to push metrics: {}", e);
+            }
+
+            thread::sleep(interval);
+        })
+        .unwrap();
+
+    Some(handler)
+}
+
+pub fn dump_metrics() -> String {
+    let mut buffer = vec![];
+    let encoder = TextEncoder::new();
```

Comment:
should support Protobuf Encoder, I worry that Prometheus may send a protobuf content request.

Reply:
From [the documentation](https://github.com/prometheus/docs/blob/master/content/docs/instrumenting/exposition_formats.md#format-variants-comparison), protobuf is not supported anymore after 2.0.

Reply:
Got it.